keep/api/utils/tenant_utils.py

```diff
@@ -15,6 +16,131 @@
 logger = logging.getLogger(__name__)
 
 
+def delete_api_key_internal(
+    session: Session,
+    tenant_id: str,
+    unique_api_key_id: str,
+) -> str:
+    """
+    Deletes API key for the given tenant.
+
+    Args:
+        session (Session): _description_
+        tenant_id (str): _description_
+        unique_api_key_id (str): _description_
+
+    Returns:
+        str: _description_
+    """
+    # Find API key
+    statement = (
+        select(TenantApiKey)
+        .where(TenantApiKey.reference_id == unique_api_key_id)
+        .where(TenantApiKey.tenant_id == tenant_id)
+    )
+
+    api_key = session.exec(statement).first()
+
+    if api_key:
+        # Delete from database
+        session.delete(api_key)
```

Comment:
don't delete but add a "is_deleted" column and change it to true. the reason is that we want to be able to keep track of the alerts that were written by API key that was deleted

Reply:
fix: f72cdc80c1540a13e0d365ca0c5f59b9e59dabbe